modular_skyrat/modules/modular_items/lewd_items/code/lewd_helpers/species.dm

```diff
@@ -0,0 +1,74 @@
+	var/atom/movable/screen/alert/aroused_x/arousal_alert = target_human.alerts["aroused"]
+	if(target_human.client?.prefs?.read_preference(/datum/preference/toggle/erp/sex_toy))
+		switch(target_human.arousal)
+			if(-INFINITY to AROUSAL_MINIMUM_DETECTABLE)
+				target_human.clear_alert("aroused", /atom/movable/screen/alert/aroused_x)
+				if(target_human.arousal < AROUSAL_MINIMUM)
+					target_human.arousal = AROUSAL_MINIMUM // To prevent massively negative values that break the lewd system for some.
+			if(AROUSAL_MINIMUM_DETECTABLE to AROUSAL_LOW)
+				throw_arousalalert("arousal_small", arousal_alert, target_human)
+			if(AROUSAL_LOW to AROUSAL_MEDIUM)
+				throw_arousalalert("arousal_medium", arousal_alert, target_human)
+			if(AROUSAL_HIGH to AROUSAL_AUTO_CLIMAX_THRESHOLD)
+				throw_arousalalert("arousal_high", arousal_alert, target_human)
+			if(AROUSAL_AUTO_CLIMAX_THRESHOLD to INFINITY) //to prevent that 101 arousal that can make icon disappear or something.
+				throw_arousalalert("arousal_max", arousal_alert, target_human)
+				if(target_human.arousal > AROUSAL_LIMIT)
+					target_human.arousal = AROUSAL_LIMIT // To prevent massively high values that cause the permahorny(tm) bug.
+
+		if(target_human.arousal > AROUSAL_MINIMUM_DETECTABLE)
+			switch(target_human.pain)
+				if(-INFINITY to AROUSAL_MINIMUM_DETECTABLE) //to prevent same thing with pain
+					arousal_alert?.cut_overlay(arousal_alert.pain_overlay)
+					if(target_human.pain < AROUSAL_MINIMUM)
+						target_human.pain = AROUSAL_MINIMUM // To prevent massively negative values that break the lewd system for some.
+				if(AROUSAL_MINIMUM_DETECTABLE to AROUSAL_LOW)
+					overlay_pain("small", arousal_alert)
+				if(AROUSAL_LOW to AROUSAL_MEDIUM)
+					overlay_pain("medium", arousal_alert)
+				if(AROUSAL_HIGH to AROUSAL_AUTO_CLIMAX_THRESHOLD)
+					overlay_pain("high", arousal_alert)
+				if(AROUSAL_AUTO_CLIMAX_THRESHOLD to INFINITY)
+					overlay_pain("max", arousal_alert)
+					if(target_human.pain > AROUSAL_LIMIT)
+						target_human.pain = AROUSAL_LIMIT
+
+		if(target_human.arousal > 1)
+			switch(target_human.pleasure)
+				if(-INFINITY to AROUSAL_MINIMUM_DETECTABLE) //to prevent same thing with pleasure
+					arousal_alert?.cut_overlay(arousal_alert.pleasure_overlay)
+				if(AROUSAL_MINIMUM_DETECTABLE to AROUSAL_LOW)
+					overlay_pleasure("small", arousal_alert)
+				if(AROUSAL_LOW to AROUSAL_MEDIUM)
+					overlay_pleasure("medium", arousal_alert)
+				if(AROUSAL_HIGH to AROUSAL_AUTO_CLIMAX_THRESHOLD)
+					overlay_pleasure("high", arousal_alert)
+				if(AROUSAL_AUTO_CLIMAX_THRESHOLD to INFINITY)
+					overlay_pleasure("max", arousal_alert)
+					if(target_human.pleasure > AROUSAL_LIMIT)
+						target_human.pleasure = AROUSAL_LIMIT
+		else
+			if(arousal_alert?.pleasure_level in list("small", "medium", "high", "max"))
+				arousal_alert.cut_overlay(arousal_alert.pleasure_overlay)
+			if(arousal_alert?.pain_level in list("small", "medium", "high", "max"))
```

Comment:
Those two lists should be just one list with all of those values in there.

Reply:
Also it probably should be numbers rather than strings, in `pain_level` and `pleasure_level`, unless those strings are really necessary.

Reply:
Used to set icon_states in other code, so somewhat necessary.